velox/functions/prestosql/tests/ArrayCumSumTest.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <optional>
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using namespace facebook::velox::functions::test;
+
+namespace {
+class ArrayCumSumTest : public FunctionBaseTest {
+ protected:
+  template <typename T>
+  void testArrayCumSum(
+      const VectorPtr& expected,
+      const VectorPtr& input,
+      const TypePtr type = CppToType<T>::create()) {
+    if (type->isDecimal()) {
```

Comment:
Is this check needed ? The input is never decimal. In any case can parseDecimalAsDouble always be false ?

Reply:
Thanks for pointing it out, reverted it. This was added for the decimal tests before.